### PR TITLE
bluetooth: controller: Use ISR_DIRECT_DECLARE for ble controller ISRs

### PR DESCRIPTION
**Pull request by rugeGerritsen**
When using interrupts with ISR_DIRECT_CONNECT(), ISR_DIRECT_DECLARE()
should be used.

Signed-off-by: Rubin Gerritsen <Rubin.Gerritsen@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/hci_driver.c [hunk 1]
```diff
@@ -505,6 +505,30 @@ static int ble_enable(void)
 	return 0;
 }
 
+ISR_DIRECT_DECLARE(ble_controller_radio_isr_wrapper)
+{
+	ble_controller_RADIO_IRQHandler();
+
+	ISR_DIRECT_PM();
+	return 1; /* We should check if scheduling decision should be made */
```
Comment (joerchan) [inline subsys/bluetooth/controller/hci_driver.c]: Are you sure about this? The direct ISRs don't use any Zephyr RTOS functionality, so they don't affect scheduling at all. The don't unblock threads at all.

**Comment by joerchan** [comment]
@rugeGerritsen Do you wish to include MPSL ISRs in this one?

File: subsys/bluetooth/controller/hci_driver.c [hunk 2]
```diff
@@ -505,6 +505,33 @@ static int ble_enable(void)
 	return 0;
 }
 
+ISR_DIRECT_DECLARE(ble_controller_radio_isr_wrapper)
+{
+	ble_controller_RADIO_IRQHandler();
+
+	ISR_DIRECT_PM();
+	return 0; /* No need for rescheduling, because the interrupt handler
```
Comment (joerchan) [inline subsys/bluetooth/controller/hci_driver.c]: Should probably put the block comment above.

Also coding standard says to do this:
```
/* multiline 
 * block comment
 */
```

**Comment by rugeGerritsen** [comment]
> @rugeGerritsen Do you wish to include MPSL ISRs in this one?

That will be handled by https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1571/files

**Comment by joerchan** [comment]
CC: @grochu @rakons @KAGA164 @kapi-no Could you give this a look to confirm nothing breaks with this change?

**Pull request merged**